flyrs/tests/test_rust_remote.py

```diff
@@ -0,0 +1,57 @@
+import os
+import time
+
+from mock import MagicMock
+
+from flytekit import task
+from flytekit.configuration import Config, SerializationSettings
+from flytekit.remote import FlyteRemote
+from flytekit.remote.entities import FlyteTask
+
+PROJECT = "flytesnacks"
+DOMAIN = "development"
+
+TASK_NAME = "test_rust_remote.my_test_task"
+TS = hash(time.time())
+VERSION_ID = f"{TS}"  # we use current timestamp when initialize tests, to prevent identical re-register.
+
+
+def test_register_task():
+    @task()
+    def my_test_task(n: int) -> int:
+        return n
+
+    remote_rs = FlyteRemote(Config.auto(), default_project=PROJECT, default_domain=DOMAIN, enable_rust=True)
+    mock_image_config = MagicMock(default_image=MagicMock(full="fake-cr.io/image-name:tag"))
+    flyte_task = remote_rs.register_task(
+        entity=my_test_task,
+        serialization_settings=SerializationSettings(image_config=mock_image_config),
```

Comment:
```suggestion
        serialization_settings=SerializationSettings(image_config=ImageConfig(default_image="flyte-cr.io/image-name:tag")),
```

Reply:
Encounter error `AttributeError: 'str' object has no attribute 'full'`,

<img width="1231" alt="Screenshot 2024-04-30 at 3 46 37 AM" src="https://github.com/flyteorg/flytekit/assets/35886692/32aff47a-98ae-46c1-97fa-6ef37bbbaace">

Reply:
try this
```
ImageConfig.auto(img_name="flyte-cr.io/image-name:tag")
```

Reply:
done, thx